### DISCUSSION TITLE
What is the component cost to build this machine?

### DISCUSSION THREAD

**Discussion by Telepresent** [dukelec/cd_pnp#9] [General]
How can I build one?

**Comment by dukelec** [comment]
I'm very happy that you are interested in CDPnP, I just wrote an instructional article on DIY CDPnP, the content still needs to be improved:
https://github.com/dukelec/cdpnp/discussions/10

Currently, assembled CDPnP is priced at ~~650USD~~ 580USD because small production runs can spread the cost, such as the cost of CNC parts.
If you do it yourself, it may be more expensive to make a small number of CNC parts unless you have your own CNC machine.

Next, we will continue to improve and optimize the design to make it more accurate and less costly (assembled expected to sell for about 500USD). And we will continue to keep open source.

**Comment by bonz099** [comment]
Hi, im every interested in purchasing one, what is the best way to contact you?

**Reply by dukelec** [reply]
Thank you for your interest, you can email me: d#d-l.io (replace # to @), or purchase directly here:
https://www.alibaba.com/product-detail/_1600819959031.html

In addition you can also contact my WhatsApp: KzE1MjAyMjI3NzEw (base64 encoded)

**Reply by SimonMerrett** [reply]
I tried sending a message on WhatsApp today but it doesn't seem to have arrived. I would be interested in purchasing hardware but also want to find out about the latest development, especially with regard to improved motion accuracy and also possible camera adjustments.
